polaris/experimental/_dataset_v2.py

```diff
@@ -0,0 +1,249 @@
+            zarr_manifest_path = generate_zarr_manifest(self.zarr_root_path, self.cache_dir)
+            self._zarr_manifest_path = zarr_manifest_path
+
+        return self._zarr_manifest_path
+
+    @computed_field
+    @property
+    def zarr_manifest_md5sum(self) -> str:
+        """
+        Lazily compute the checksum once needed.
+
+        The checksum of the DatasetV2 is computed from the Zarr Manifest and is _not_ deterministic.
+        """
+        if not self.has_zarr_manifest_md5sum:
+            logger.info("Computing the checksum. This can be slow for large datasets.")
+            self.zarr_manifest_md5sum = calculate_file_md5(self.zarr_manifest_path)
+        return self._zarr_manifest_md5sum
+
+    @zarr_manifest_md5sum.setter
+    def zarr_manifest_md5sum(self, value: str):
+        """Set the checksum."""
+        if not re.fullmatch(r"^[a-f0-9]{32}$", value):
+            raise ValueError("The checksum should be the 32-character hexdigest of a 128 bit MD5 hash.")
+        self._zarr_manifest_md5sum = value
+
+    @property
+    def has_zarr_manifest_md5sum(self) -> bool:
+        """Whether the md5sum for this dataset's zarr manifest has been computed and stored."""
+        return self._zarr_manifest_md5sum is not None
+
+    def get_data(self, row: int, col: str, adapters: dict[str, Adapter] | None = None) -> np.ndarray | Any:
+        """Indexes the Zarr archive.
+
+        Args:
+            row: The index of the data to fetch.
+            col: The label of a direct child of the Zarr root.
+            adapters: The adapters to apply to the data before returning it.
+                If None, will use the default adapters specified for the dataset.
+
+        Returns:
+            A numpy array with the data at the specified indices. If the column is a pointer column,
+                the content of the referenced file is loaded to memory.
+        """
+        # Fetch adapters for dataset and a given column
+        # Partially override if the adapters parameter is specified.
+        adapters = {**self.default_adapters, **(adapters or {})}
+        adapter = adapters.get(col)
+
+        # Get the data
+        group_or_array = self.zarr_data[col]
+
+        # If it is a group, there is no deterministic order for the child keys.
+        # We therefore use a special array that defines the index.
+        if isinstance(group_or_array, zarr.Group):
+            row = group_or_array[_INDEX_ARRAY_KEY][row]
+        arr = group_or_array[row]
+
+        # Adapt the input to the specified format
+        if adapter is not None:
+            arr = adapter(arr)
+
+        return arr
+
+    def upload_to_hub(self, access: AccessType = "private", owner: HubOwner | str | None = None):
+        """Uploads the dataset to the Polaris Hub."""
+
+        # NOTE (cwognum):  Leaving this for a later PR, because I want
+        #  to do it simultaneously with a PR on the Hub side.
+        raise NotImplementedError
+
+    def to_json(
```

Comment:
**Comment**
Not for this PR, but this method is becoming increasingly ill-named.

Reply:
Maybe `save` and `load`.